src/shims/unix/socket.rs

```diff
@@ -58,9 +58,9 @@ impl FileDescription for SocketPair {
         ecx: &mut MiriInterpCx<'tcx>,
     ) -> InterpResult<'tcx, io::Result<usize>> {
         let request_byte_size = bytes.len();
-        let data_size: usize = self.readbuf.borrow().buf.len();
-        let readbuf_has_writer = self.readbuf.borrow().buf_has_writer;
-        let mut mut_readbuf = self.readbuf.borrow_mut();
+        let mut readbuf = self.readbuf.borrow_mut();
+        let data_size: usize = readbuf.buf.len();
+        let readbuf_has_writer = readbuf.buf_has_writer;
```

Comment:
Both of these let-bindings are used exactly once. Are they worth having as separate variables here or does the code become cleaner when they get inlined?

In fact data_size is just used in a `== 0` check so that could become `if readbuf.buf.is_empty()`.